arches/app/models/models.py

```diff
@@ -1360,14 +1360,24 @@ def __init__(self, *args, **kwargs):
     def is_fully_provisional(self):
         return bool(self.provisionaledits and not any(self.data.values()))
 
+    def set_missing_keys_to_none(self):
+        any_key_set = False
+        if not self.nodegroup_id:
+            return any_key_set
+        for node in self.nodegroup.node_set.all():
+            if node.datatype == "semantic":
+                continue
+            node_id_str = str(node.pk)
+            if not node_id_str in self.data:
+                self.data[node_id_str] = None
+                any_key_set = True
+
+        return any_key_set
+
     def save(self, *args, **kwargs):
+        if any_key_set := self.set_missing_keys_to_none():
```

Comment:
doesn't look like `any_key_set` is used here:

```suggestion
        if self.set_missing_keys_to_none():
```

Reply:
I guess I thought this was more clear, since from the method name you can't tell why it would have a return value.

Reply:
I guess I assume that anything used in an "if" statement has a return value.  So that didn't bother me as much as the return value not being used.

Reply:
[78f0f52](https://github.com/archesproject/arches/pull/12138/commits/78f0f524b628454ac595d93eba208c93bf600644)